application/controllers/upload.py

```diff
@@ -10,21 +10,28 @@
     base_template_context,
     group_by_group)
 from performanceplatform.client.data_set import DataSet
+import json
 
 
 @app.route('/upload-data', methods=['GET'])
 @requires_authentication
 def upload_list_data_sets(admin_client):
     template_context = base_template_context()
+    template_context.update({
+        'user': session['oauth_user'],
+    })
     try:
         data_sets = group_by_group(admin_client.list_data_sets())
     except HTTPError as err:
         if err.response.status_code == 401:
             return redirect(url_for('oauth_sign_out'))
         else:
-            raise
+            error = "Could not retrieve list of datasets: {}".format(
```

Comment:
Will this be a bit too specific message wise for some of our users? I think it's probably okay but other opinions welcome:
<img width="637" alt="screen shot 2015-07-14 at 16 34 00" src="https://cloud.githubusercontent.com/assets/677893/8677314/40eb2dc2-2a46-11e5-9632-8acdf97c359b.png">